pkg/sources/manager.go

```diff
@@ -31,35 +34,43 @@ const (
 )
 
 var (
-	// Last time Heapster performed a scrape since unix epoch in seconds.
 	lastScrapeTimestamp = prometheus.NewGaugeVec(
 		prometheus.GaugeOpts{
-			Namespace: "heapster",
+			Namespace: "metrics_server",
 			Subsystem: "scraper",
 			Name:      "last_time_seconds",
-			Help:      "Last time Heapster performed a scrape since unix epoch in seconds.",
+			Help:      "Last time metrics-server performed a scrape since unix epoch in seconds.",
 		},
 		[]string{"source"},
 	)
 
-	// Time spent exporting scraping sources in microseconds..
-	scraperDuration = prometheus.NewSummaryVec(
-		prometheus.SummaryOpts{
-			Namespace: "heapster",
-			Subsystem: "scraper",
-			Name:      "duration_microseconds",
-			Help:      "Time spent scraping sources in microseconds.",
-		},
-		[]string{"source"},
-	)
+	// initialized below in registerScrapeDurationMetric
+	scraperDuration     *prometheus.HistogramVec
+	scraperDurationOnce sync.Once
 )
 
 func init() {
 	prometheus.MustRegister(lastScrapeTimestamp)
-	prometheus.MustRegister(scraperDuration)
+}
+
+func registerScraperDurationOnce(scrapeTimeout time.Duration) {
+	scraperDurationOnce.Do(func() {
```

Comment:
this is an anti-pattern, but I understand that the current architecture makes this hard not to do, can we put a TODO here to avoid the necessity of a sync.Once?

Reply:
what's the general alternative?  We don't really know a good buckets list until we know the timeout, but we also don't want to call this more than once.  We *should* only ever get one manager, so it shouldn't be a problem, but I figured better to put it in a `Once` than cause the program to crash over metrics issues.

Reply:
If a manager is really always only created once, then we should just remove it. Otherwise we create the metric with the appropriate buckets in the same part as what creates the manager and pass the metric to the manager.

Reply:
ack, I'll try to fix it up.

Reply:
(we can't actually assume a manager is only ever created once, because it might be created multiple times for tests)

Reply:
this look better now?

Reply:
felt mild uncomfortable passing it in explicitly, since it seems like mixing actual signature with implementation details -- the manager should function regardless of the presence or absence of the metric.